pattoo_shared/installation/environment.py

```diff
@@ -0,0 +1,58 @@
+"""Functions for setting up a virtual environment for the installation."""
+import os
+import getpass
+from pattoo_shared.installation import shared, configure
+
+
+def make_venv(file_path):
+    """Create virtual environment for pattoo installation.
+
+    Args:
+        file_path: The path to the virtual environment
+
+    Returns:
+        None
+    """
+    # Say what we're doing
+    print('??: Create virtual environment')
+    command = 'python3 -m virtualenv {}'.format(file_path)
+    shared.run_script(command)
+    print('OK: Virtual environment created')
+    # Ensure venv is owned by pattoo if the pattoo user exists
+    if configure.user_exists('pattoo') and configure.group_exists('pattoo'):
+        if getpass.getuser() == 'root':
+            shared.run_script('chown -R pattoo:pattoo {}'.format(file_path))
+
+
+def activate_venv(activation_path):
+    """Activate the virtual environment in the current interpreter.
+
+    Args:
+        activation_path: The path to the activate_this.py file
+
+    Returns:
+        None
+
+    """
+    # Open activte_this.py for reading
+    with open(activation_path) as f:
```

Comment:
I discovered this preferred way of accessing files with error handling: https://docs.python.org/3/library/os.html [Search for os.access](https://docs.python.org/3/library/os.html)


```
    try:
        fh_ = open(filename, 'rb')
    except PermissionError:
        # Do something
        log.log2(....
    except FileNotFoundError:
        # Do something
        log.log2(....
    except:
        # Do something
        log.log2(....
    else:
        with fh_:
            yaml.dump(self._config, fh_, default_flow_style=False)

```